src/app/features/sidebar/hooks/index.ts

```diff
@@ -67,35 +66,31 @@ export const useSearchHistory = () => {
   }, [historyEntries])
 }
 
-export const useQueryItemMenu = (data, tree, handlers) => {
+export const useQueryItemMenu = ({data, tree, handlers, lakeId}) => {
```

Comment:
Would you mind turning this into a thunk, so that the component doesn't need to re-render whenever one of these useSelector fires?